msdk-featdet/msdk-featdet-ADAP-3D/src/main/java/io/github/msdk/featdet/ADAP3D/common/algorithms/BiGaussian.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * (C) Copyright 2015-2017 by MSDK Development Team
+ *
+ * This software is dual-licensed under either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1 as published by the Free
+ * Software Foundation
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by the Eclipse Foundation.
+ */
+package io.github.msdk.featdet.ADAP3D.common.algorithms;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.lang.Math;
+
+import org.apache.commons.collections4.MapIterator;
+import org.apache.commons.collections4.map.MultiKeyMap;
+
+import io.github.msdk.featdet.ADAP3D.common.algorithms.SliceSparseMatrix.Triplet;
+
+import org.apache.commons.collections4.keyvalue.MultiKey;
+
+/**
+ * <p>
+ * BiGaussian Class is used for fitting BiGaussian on EIC.
+ * BiGaussian is composed of 2 halves of Gaussian with different standard deviations.
+ * It depends on 4 parameters (height, mu, sigmaLeft, sigmaRight) and computed by the formula
+ * </p>
+ * 
+ * <p>f(x) = height * exp(-(x-mu)^2 / (2 * sigmaRight^2)) if x > mu</p>
+ * <p>f(x) = height * exp(-(x-mu)^2 / (2 * sigmaLeft^2)) if x < mu</p>
+ */
+public class BiGaussian {
+	
+	//This is used to store horizontal slice of sparse matrix.
+	private final MultiKeyMap<Integer, Triplet> horizontalSlice;
+	
+	//This is used to store all the intensities of slice. 
+	private final List<Double> listOfIntensities;
+	
+	enum Direction {RIGHT, LEFT}
+	
+	//This is used for storing BiGaussian parameters inside constructor.	
+	private static class BiGaussianParams{
+		public static double maxHeight;
+		public static int mu;
+		public static double sigmaLeft;
+		public static double sigmaRight;
+	}
+	
```

Comment:
I think making these variables static is dangerous. What happens if two difference instances of this class are used by different threads?

I suggest the following, assuming @tomas-pluskal agrees:

```
private static class BiGaussianParams{
  public final double maxHeight;
  public final int mu;
  public final double sigmaLeft;
  public final double sigmaRight;

  public BiGaussianParams(double maxHeight, int mu, double sigmaLeft, double sigmaRight) {
    this.maxHeight = maxHeight;
    this.mu = mu;
    this.sigmaLeft = sigmaLeft;
    this.sigmaRight = sigmaRight;
  }
}
```